tdct/deconvolution.py

```diff
@@ -0,0 +1,864 @@
+        data_deconv_uint8 = data_deconv_norm_256.astype('uint8')
+        
+        progr0.increment() #5
+
+        #Save data
+        fpath,fname = os.path.split(datapath)
+        fname0 = os.path.join(fpath, "DeconvRL"+str(niter)+"it_"+fname)
+        tf.imsave(fname0 ,data_deconv_uint8)
+
+        progr0.setmax()
+ 
+        if debug is True: print(clrmsg.DEBUG, "Completed deconvolution, file saved: ", fname0)
+
+def doRLDeconvolution5(datapath , psfdatapath , niter=0, qtprocessbar=None):
+    '''Deconvolute data using the Richardson Lucy algorithm.
+    This code is based on the source code for DeconvolutionLab2
+    https://github.com/Biomedical-Imaging-Group/DeconvolutionLab2/blob/master/src/main/java/deconvolution/algorithm/RichardsonLucy.java
+    Instead of doing convolutions using raw fft routines, Here we use convolution algorithms in scipy.signal.convolve and scipy.signal.correlate
+
+    This is working well and appears to give same results as DeconvolutionLab2
+    '''
+
+    #Estimate progress iterations
+    nProgrIter = 2*niter + 6 #Check if ok
+    progr0 = _progrBarHandle(qtprocessbar, nProgrIter) #Sets up
+
+    #Read data
+    if os.path.isfile(datapath) is True and os.path.isfile(psfdatapath) is True and niter>0:
+        if debug is True: print(clrmsg.DEBUG, "Loading images: ", datapath," , ",psfdatapath)
+        progr0.increment() #1
+
+        data_np = tf.imread(datapath)
+        psf_np = tf.imread(psfdatapath)
+
+        progr0.increment() #2
+
+        bestshape = data_np.shape
+        
+        #Normalise
+        data_np_norm = _convertAndNormalise(data_np)
+        psf_np_norm = _convertAndNormalise(psf_np)
+
+        progr0.increment() #3
+
+        xn1 = np.array(data_np_norm) #initialize copy
+
+        progr0.increment() #4
+
+        for i in range(niter):
+            xn=xn1
+
+            Hx = scipy.signal.convolve(xn, psf_np,mode='same', method='fft')
+
+            yhx = np.divide(data_np_norm,Hx)
+            
+            progr0.increment()
+
+            #correlation of the result with psf (note that is not a convolution)
+            htyhx = scipy.signal.correlate(yhx, psf_np, mode='same', method='fft')
+
+            xn1 = np.multiply(xn, htyhx)
+            
+            progr0.increment()
+
+        #At the end xn1 should have the deconvoluted data
+        
+        #Convert to uint8
+        data_deconv_norm_256 = _convertAndNormalise(xn1)*256
+        data_deconv_uint8 = data_deconv_norm_256.astype('uint8')
+        
+        progr0.increment() #5
+
+        #Save data
+        fpath,fname = os.path.split(datapath)
+        fname0 = os.path.join(fpath, "DeconvRL"+str(niter)+"it_"+fname)
+        tf.imsave(fname0 ,data_deconv_uint8)
+
+        progr0.setmax()
+ 
+        if debug is True: print(clrmsg.DEBUG, "Completed deconvolution, file saved: ", fname0)
+
+
+def doRLDeconvolution7(datapath , psfdatapath , niter=0, qtprocessbar=None):
```

Comment:
Ok, staring my review again from here. Again, try to use python convention with names and doc strings.